edb/server/pgcon/pgcon.pyx

```diff
@@ -1667,13 +1667,21 @@ cdef class PGConnection:
                             field_type = self.buffer.read_byte()
                             if field_type == b'P':  # Position
                                 msg_buf.write_byte(b'q')  # Internal query
-                                msg_buf.write_bytestring(query)
+                                msg_buf.write_bytestring(unit.input_query.encode('utf8'))
```

Comment:
After translation, we don't need to include the original query with "internal query" (`q`) any more - we can simply return the translated data with the "position" (`P`), instead of the "internal position" (`p`). I believe psql already has the SQL and it could produce a proper error output.